File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* findArray(int* pref, int prefSize, int* returnSize) {
    int *ans = (int*)malloc(prefSize * sizeof(int));
    *returnSize = prefSize;
    ans[0] = pref[0];
    for (int i = 0; i < prefSize - 1; i++) {
        ans[i+1] = pref[i] ^ pref[i + 1];
    }
    return ans;
} 

File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.py
class Solution:
    def findArray(self, pref: List[int]) -> List[int]:
        if len(pref) == 0 or len(pref) == 1:
            return pref
        
        else:
            ans = [pref[0]]
            for i in range(len(pref)-1):
                ans.append(pref[i] ^ pref[i+1])
            return ans